### ISSUE TITLE
NormalizeDate manipulator should accommodate dd/mm/yyyy and mm/dd/yyyy

### ISSUE THREAD

**Issue by mjordan** [MarcusBarnes/mik#195]
The regex `/^(\d\d)\-(\d\d)\-(\d\d\d\d)$` in the NormalizeDate metadata manipulator is intended to match on dd/mm/yyyy. This pattern is identical to the one that would match mm/dd/yyyy. We should add a parameter to indicate which we want to match on.

**Comment by mjordan** [comment]
I've added a third, optional parameter, 'm', which indicates that for patterns that can be interpreted as either starting with a day or a month, the first part of the incoming date values should be considered months. For example,

`metadatamanipulators[] = "NormalizeDate|Date|dateCreated|m"`

will indicate that the first part of dates in the incoming metadata like 10/9/1923 and 10-9-1923 is a month. The absence of the third parameter or any other value than 'm' will tell NormalizeDate to interpret the first part of the string as a day.

I've added some PHPUnit tests and they pass. This new feature is most easily smoke tested using CSV input data, where you'd test first with dates without the 'm' parameter in dd-mm-yyy and dd/mm/yyyy format, then change your dates in your input data and rerun the job with the 'm' parameter. In both cases, the dates in your output MODS should be in yyyy-mm-dd format. 

@MarcusBarnes  I'd like to use this for a current migration job so I'm OK with merging it into master without having you test if you are.
